src/Ccap/Codegen/Scala.purs

```diff
@@ -127,17 +135,25 @@ typeDecl (TypeDecl name tt an) =
             text "type" <<+>> text name <<+>> char '=' <<+>> text typ
               // wrapEncoder name t (text encode)
               // wrappedDecoder
-    Record props ->
+    Record props -> do
+      Module modName _ _ <- asks _.currentModule
       let
-        cls = (text "final case class" <<+>> text name) `paren` (recordFieldType <$> props)
+        qualify =
+          if modName == name && outputMode == TopLevelCaseClass
```

Comment:
How would that look syntactically? It seems like we meet ourselves coming and going one way or another here since any syntax that's aligns with Scala's way of doing things won't align with PureScript and vice versa.